### PR TITLE
Add Single.fromCallable()

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Second part of #3154.

@akarnokd I'd like to reuse `Observable.fromCallable()` but didn't find apropriate way to do it: `Observable.fromCallable().toSingle()` is not best option, if you have some thoughts on this — please share them.

### CODE REVIEW

**Comment by akarnokd** [comment]
:+1: I'm not too keen on reusing but the most complicated operators for `Single`.

File: src/main/java/rx/Single.java [hunk 1]
```diff
@@ -605,6 +606,42 @@ public void call(SingleSubscriber<? super T> te) {
...
+     *   <dd>{@code fromCallable} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     *
+     * @param func
+     *         function which execution should be deferred, it will be invoked when Observer will subscribe to the {@link Single}.
+     * @param <T>
+     *         the type of the item emitted by the {@link Single}.
+     * @return a {@link Single} whose {@link Observer}s' subscriptions trigger an invocation of the given function.
+     */
+    public static <T> Single<T> fromCallable(final Callable<? extends T> func) {
```
Comment (akarnokd) [inline src/main/java/rx/Single.java]: This should start out as experimental unless the RxJava contributors want to fast-track this.

Comment (artem-zinnatullin) [inline src/main/java/rx/Single.java]: Added `@Experimental`

**Comment by abersnaze** [comment]
@artem-zinnatullin rebase please.

**Comment by artem-zinnatullin** [comment]
@abersnaze rebased, I'll rebase one PR at a time, since they're all conflict with each other.

**Pull request merged**